01-messaging.md

```diff
@@ -82,6 +83,79 @@ however, adding a 6-byte padding after the type field was considered
 wasteful: alignment may be achieved by decrypting the message into
 a buffer with 6-bytes of pre-padding.
 
+## Wire Type-Length-Value Format
+
+Throughout the protocol, we use a TLV (Type-Length-Value) format to allow for
+the backwards-compatible addition of new fields to existing message types. This
+format describes the TLV encoding specific to wire messages.
+
+A `tlv_record` represents a single field, encoded in the form:
+
+* [`1`: `type`]
+* [`varint`: `length`]
+* [`length`: `value`]
+
+A `tlv_stream` is a series of (possibly zero) `tlv_record`s, represented as the
+concatenation of the encoded `tlv_record`s. When used to extend existing
+messages, a `tlv_stream` is typically placed after all currently defined fields.
+
+The `type` is a big-endian 8-bit identifier for the `tlv_record` and determines
+how the contents of `value` should be decoded.
+
+The `length` is a varint encoded using the Bitcoin `CompactSize` format to
+signal the size of `value` in bytes.
+
+The `value` depends entirely on the `type`. Each `tlv_stream` has its own 8-bit
+namespace for defining the meaning of its `type` identifiers.
+
+### Requirements
+
+The sending node:
+ - MUST order `tlv_record`s in a `tlv_stream` by monotonically-increasing `type`.
+ - SHOULD NOT use the varbytes encoding for `value`s containing byte arrays.
+
+The receiving node:
+ - if encounters EOF while parsing a `type`:
+   - MUST stop parsing the `tlv_stream`.
+ - if encounters EOF while parsing a `length` or `value`:
+   - MUST fail to parse to `tlv_stream`.
+ - if `type` and `length` parse:
+   - if `type` is known:
+     - MUST decode the next `length` bytes using the known encoding for `type`.
+   - otherwise, if `type` is unknown:
+       - MUST discard the next `length` bytes.
+ - if the `type`s in a `tlv_stream` are not monotonically-increasing:
+   - MUST fail to parse the `tlv_stream`.
+
+### Rationale
+
+The primary advantage in using TLV is that a reader is able to ignore new fields
+that it does not understand, since each field carries the exact size of the
+encoded element. Without TLV, even if a node does not wish to use a particular
+field, the node is forced to add parsing logic for that field in order to
+determine the offset of any fields that follow.
+
+The monotonicity constraint ensures that all `type`s are unique and can appear
+at most once.
+
+The use of a varint for `length` permits a space savings over a fixed 16-bit
+`length` for `value`s whose encode is less than 253 bytes. When `tlv_stream` is
+used in EOB multi-frame payload, this can potentially save payloads from
```

Comment:
You're right, I don't think it is. I ended up removing it and just referring to them as multi-frame payloads, since this better matches the language in #593 

2583ae291eaa672f12da4a35f057ee1f48f71871